src/test/java/io/vertx/core/net/NetTest.java

```diff
@@ -3368,6 +3394,104 @@ public void testHalfCloseCallsEndHandlerAfterBuffersAreDelivered() throws Except
     await();
   }
 
+  @Test
+  public void testSslHandshakeTimeoutHappened() throws Exception {
+    server.close();
+    client.close();
+
+    // set up a normal server to force the SSL handshake time out in client
+    NetServerOptions serverOptions = new NetServerOptions()
+      .setSsl(false);
+    server = vertx.createNetServer(serverOptions);
+
+    NetClientOptions clientOptions = new NetClientOptions()
+      .setSsl(true)
+      .setTrustAll(true)
+      .setSslHandshakeTimeout(200)
+      .setSslHandshakeTimeoutUnit(TimeUnit.MILLISECONDS);
+    client = vertx.createNetClient(clientOptions);
+
+    server.connectHandler(s -> {
+    }).listen(testAddress, ar -> {
+      assertTrue(ar.succeeded());
+      client.connect(testAddress, res -> {
```

Comment:
use `onFailure` construct for this test:

```
client.connect(testAddress, onFailure(err -> { ... });
```

this avoids to test the async result and gives directly the error cause